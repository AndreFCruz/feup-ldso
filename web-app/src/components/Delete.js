import React, {Component} from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

class Delete extends Component{

    constructor(props){
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event){
        event.preventDefault();

        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects?where[id]=' + this.props.match.params.project;
        fetch(route, {
            method: 'DELETE',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            }
        });
    }

    render(){
        return(
            <Button color="secondary" onClick={this.handleSubmit}>Delete</Button>
        );
    }
}

Delete.propTypes = {
    match: PropTypes.object.isRequired
};


export default Delete;
