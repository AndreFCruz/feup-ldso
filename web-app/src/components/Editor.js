import React, {Component} from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

class Editor extends Component {

    constructor(props) {
        super(props);

        this.state = {
            error: null,
            isLoaded: false,
            project: []
        };

        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects?filter[where][id]=' + this.props.match.params.project;
        fetch(route)
            .then(res => res.json())
            .then(
                (result) => {
                    this.setState({
                        isLoaded: true,
                        project: result,
                    });
                },
                (error) => {
                    this.setState({
                        isLoaded: true,
                        error
                    });
                }
            );
    }

    handleInputChange(event) {
        this.setState({
            [event.target.name]: event.target.value
        });
    }
    //o url com filtro fica .../social-projects?filter[where][id]=<id_here>

    handleSubmit(event){
        event.preventDefault();
        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects?where[id]=' + this.props.match.params.project;
        fetch(route, {
            method: 'PATCH',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(this.state),
        })
            .then(response => response.json());
    }

    render() {
        const { project } = this.state;

        return (
            <div>
                {
                    project.map(proj => (
                        <form key={proj.id}>
                            <label>
          Project Title
                                <br/>
                                <input
                                    name="title"
                                    placeholder={proj.title}
                                    value={this.state.title}
                                    onChange={this.handleInputChange}/>
                            </label>
                            <br/>
                            <label>
          Project Description
                                <br/>
                                <textarea
                                    name="content"
                                    placeholder={proj.content}
                                    value={this.state.content}
                                    onChange={this.handleInputChange}/>
                            </label>
                            <br/>
                            <label>
                                <Button color="secondary" onClick={this.handleSubmit}>Edit</Button>
                            </label>
                        </form>
                    ))}
            </div>
        );}
}

Editor.propTypes = {
    match: PropTypes.object.isRequired
};

export default Editor;
