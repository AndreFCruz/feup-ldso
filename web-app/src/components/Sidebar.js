import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { Button, ListGroupItem, ListGroup } from 'reactstrap';

class Sidebar extends Component {
    constructor(props) {
        super(props);

        this.state = {
            error: null,
            isLoaded: false,
            projects: []
        };

    }

    componentDidMount() {
        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects';
        fetch(route)
            .then(res => res.json())
            .then(
                (result) => {
                    this.setState({
                        isLoaded: true,
                        projects: result,
                    });
                },
                (error) => {
                    this.setState({
                        isLoaded: true,
                        error
                    });
                }
            );

    }

    componentDidUpdate(prevProps) {
        const route = process.env.REACT_APP_ENDPOINT + 'faculties/en/' + this.props.match.params.faculty.toLowerCase() + '/social-projects';
        if ((prevProps.match.params.faculty !== this.props.match.params.faculty) || this.props.onGetRefresh()){
            fetch(route)
                .then(res => res.json())
                .then(
                    (result) => {
                        this.setState({
                            isLoaded: true,
                            projects: result,
                        });
                    },
                    (error) => {
                        this.setState({
                            isLoaded: true,
                            error
                        });
                    }
                );

            this.props.onChildUnsetRefresh();
        }
    }

    render() {
        const { error, isLoaded, projects } = this.state;
        if (error) {
            return <div>Error: {error.message}</div>;
        } else if (!isLoaded) {
            return <div>Loading...</div>;
        } else {
            return (
                <div>
                    <NavLink to={'/faculties/' + this.props.match.params.faculty.toLowerCase()}>
                        <Button color="secondary">Add Project</Button>
                    </NavLink>
                    <ListGroup>
                        {
                            projects.map(proj => (
                                <ListGroupItem key={proj.id} className="faculty_proj">
                                    <NavLink to={'/faculties/' + this.props.match.params.faculty.toLowerCase() + `/${proj.id}`}>{proj.title}</NavLink>
                                </ListGroupItem>
                            ))}
                    </ListGroup>
                </div>
            );
        }
    }
}


Sidebar.propTypes = {
    match: PropTypes.object.isRequired,
    onChildUnsetRefresh: PropTypes.func.isRequired,
    onGetRefresh: PropTypes.func.isRequired
};

export default Sidebar;
