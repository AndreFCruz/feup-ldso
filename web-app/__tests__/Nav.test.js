import React from 'react';
import rendered from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Sidebar from '../src/components/Sidebar';
import Delete from '../src/components/Delete';
import Editor from '../src/components/Editor';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';

import 'isomorphic-fetch'; //components use fetch() on mount

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

var mockAxios = new MockAdapter(axios);

var props = {
    match: {
        params: {
            faculty: 'faup',
            project: '1'
        }
    }
};

it('renders elements', () => {

    mockAxios.onGet('http://localhost:3000/faculties/en/faup/social-projects').reply(200, {
        title: '[EN] Sed in convallis nulla',
        short_description: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Etiam neque. Aliquam erat volutpat. Suspendisse sagittis ultrices augue. Pellentesque ipsum.',
        content: '[EN] Aliquam sit amet varius lacus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Sed tortor justo, ornare a vulputate in, mattis id mi. Donec varius, dui venenatis pharetra ultricies, ligula nulla ornare ex, quis mattis libero quam a tellus. Quisque quis est id nulla ornare semper at a nisi. Integer tincidunt egestas ornare. Nullam congue tellus arcu, eu ultricies mi pellentesque eget. Pellentesque massa sapien, suscipit sit amet lacus vel, faucibus faucibus elit. Cras in metus viverra, luctus magna ac, tristique leo. Aenean ultrices dui eget velit cursus, vitae suscipit purus varius. Aliquam erat volutpat. In porta metus et velit aliquet porta.',
        start_date: '2017-01-01',
        images: [
            'https://picsum.photos/200/300/?random',
            'https://picsum.photos/112/200/?random'
        ],
        facultyId: 1
    });
    const mockupCall = jest.fn();

    const sidebarComponent = rendered.create(
        <BrowserRouter>
            <Sidebar
                {...props}
                OnChildSetRefresh={mockupCall}
                onGetRefresh={mockupCall}
                onChildUnsetRefresh={mockupCall}
            />
        </BrowserRouter>);
    let tree = sidebarComponent.toJSON();
    expect(tree).toMatchSnapshot();

    const editorComponent = rendered.create(
        <BrowserRouter>
            <Editor
                {...props}
                onChildSetRefresh={mockupCall}
            />
        </BrowserRouter>
    );
    let treeList = editorComponent.toJSON();
    expect(treeList).toMatchSnapshot();

    const deleteComponent = rendered.create(
        <BrowserRouter>
            <Delete
                {...props}
                onChildSetRefresh={mockupCall}
            />
        </BrowserRouter>
    );
    let treeForm = deleteComponent.toJSON();
    expect(treeForm).toMatchSnapshot();
});
