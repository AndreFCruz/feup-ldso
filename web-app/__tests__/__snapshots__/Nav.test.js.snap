// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders elements 1`] = `
<div>
  Loading...
</div>
`;

exports[`renders elements 2`] = `<div />`;

exports[`renders elements 3`] = `
<button
  aria-label={null}
  className="btn btn-secondary"
  onClick={[Function]}
  type="button"
>
  Delete
</button>
`;

exports[`renders elements 4`] = `
<div
  className="menu"
>
  <div
    className="card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180"
      width="100%"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Manage Social Impact Projects
      </h5>
      <p
        className="card-text"
      >
        Some quick example text to build on the card title and make up the bulk the cards content
      </p>
      <a
        aria-current={null}
        href="/faculties/faup/socialProjects"
        onClick={[Function]}
      >
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
        >
          Go
        </button>
      </a>
    </div>
  </div>
  <div
    className="card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180"
      width="100%"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Manage Promotional Videos
      </h5>
      <p
        className="card-text"
      >
        Some quick example text to build on the card title and make up the bulk the cards content
      </p>
      <a
        aria-current={null}
        href="/faculties/faup/promotionalVideos"
        onClick={[Function]}
      >
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
        >
          Go
        </button>
      </a>
    </div>
  </div>
  <div
    className="card"
  >
    <img
      alt="Card image cap"
      className="card-img-top"
      src="https://placeholdit.imgix.net/~text?txtsize=33&txt=318%C3%97180&w=318&h=180"
      width="100%"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Manage Future Prospects
      </h5>
      <p
        className="card-text"
      >
        Some quick example text to build on the card title and make up the bulk the cards content
      </p>
      <a
        aria-current={null}
        href="/faculties/faup/futureProspects"
        onClick={[Function]}
      >
        <button
          aria-label={null}
          className="btn btn-secondary"
          onClick={[Function]}
        >
          Go
        </button>
      </a>
    </div>
  </div>
</div>
`;
